torch/quantization/fx/quantization_patterns.py

```diff
@@ -1309,9 +1320,30 @@ def convert(self,
                 quantized_graph: Graph,
                 node_name_to_scope: Dict[str, Tuple[str, type]],
                 load_arg: Callable,
+                env: Dict[str, Dict[Optional[torch.dtype], Node]],
                 is_reference: bool = False,
                 convert_custom_config_dict: Dict[str, Any] = None) -> Node:
-        return quantized_graph.node_copy(node, load_arg(quantized=None))
+        if is_reference:
+            activation_post_process = \
+                self._maybe_get_last_node_only_observer(modules)
+            # when there is no activation_post_process following the CopyNode, it means
+            # that the CopyNode is configured with a qconfig that doesnot require
+            # observation, e.g. dynamic_qconfig
+            if activation_post_process is None:
+                op_out = quantized_graph.node_copy(node, load_arg(quantized=torch.float))
+                return op_out
+            else:
+                args = load_arg(quantized=[torch.quint8])(node.args)
+                args = list(load_arg(quantized=torch.float)(node.args))
+                kwargs = load_arg(quantized=torch.float)(node.kwargs)
+                op_out = quantized_graph.node_copy(node, load_arg(quantized=torch.float))
+                env[node.name][torch.float] = op_out
```

Comment:
is env actually used inside of `quantize_node`, i.e. would things break if we returned a list of nodes and put them in `env` in the caller?

accepting to unblock